sdk/src/main/java/com/mapbox/maps/renderer/FpsManager.kt

```diff
@@ -0,0 +1,208 @@
+package com.mapbox.maps.renderer
+
+import android.os.Handler
+import android.os.Looper
+import androidx.annotation.AnyThread
+import androidx.annotation.WorkerThread
+import androidx.core.os.postDelayed
+import com.mapbox.maps.logE
+import com.mapbox.maps.logI
+import com.mapbox.maps.logW
+import kotlin.math.pow
+
+internal class FpsManager {
+  private var userDefinedRefreshRate = USER_DEFINED_REFRESH_RATE_NOT_SET
+  private var userDefinedRatio = USER_DEFINED_RATIO_NOT_SET
+
+  private var screenRefreshRate = SCREEN_METRICS_NOT_DEFINED
+  private var screenRefreshPeriodNs = -1L
+  private var previousRefreshNs = -1L
+  private var previousDrawnFrameIndex = 0
+  private var frameRenderTimeAccumulatedNs = 0L
+
+  private var preRenderTimeNs = -1L
+  private var choreographerTicks = 0
+  private var choreographerSkips = 0
+
+  @Volatile
+  private var fpsChangedListener: OnFpsChangedListener? = null
+
+  private var handler: Handler? = null
+
+  @WorkerThread
+  fun prepareHandler() {
+    Looper.myLooper()?.let { looper ->
```

Comment:
Is it valid case at all, renderThread without Looper? Shouldn't we throw an exception?

Reply:
Also - maybe just pass prepared handler from outside?

Reply:
> Is it valid case at all, renderThread without Looper? Shouldn't we throw an exception?

no, guess it's not but my idea is that `FpsManager` is generally pretty rarely used class and I don't want it be the root cause for runtime exceptions for whole render thread

 > Also - maybe just pass prepared handler from outside?

passing handler from the outside requires this handler be created while I did design `FpsManager` been created in render thread ctor so that even if user does pass max FPS value _before_ render thread instance is indeed prepared - it will be also working as expected

Reply:
refactored it all